src/App.js

```diff
@@ -117,16 +122,128 @@ class App extends React.Component {
     }
   }
 
-  async initAuthorityLists () {
-    this.data.authorityOriginData = await util.getAuthorityLists(
+  async initContractData() {
+    await this.getAuthorityData()
+    await this.initBallotData()
+    util.setUpdateTimeToLocal(new Date())
+  }
+
+  async refreshContractData(forced=false) {
+    const updateTime = forced ? 0 : util.getUpdateTimeFromLocal.value
+    if(updateTime + 300000 > Date.now()) return
+    await this.getAuthorityData()
+    await this.getBallotData()
+    await this.modifyBallotData()
+    util.setUpdateTimeToLocal(new Date())
+  }
+
+  async getAuthorityData() {
+    const modifiedBlock = await contracts.governance.govInstance.methods.modifiedBlock().call() // need to edit contract
+    if(modifiedBlock === util.getModifiedFromLocal() && util.getAuthorityFromLocal()) {
+      this.data.authorityOriginData = util.getAuthorityFromLocal()
+      return
+    }
+    await this.initAuthorityData()
+    util.setModifiedToLocal(modifiedBlock)
+  }
+
+  async getBallotData() {
+    const ballotCnt = await contracts.governance.getBallotLength()
+    let localBallotCnt = Object.keys(this.data.ballotBasicOriginData).length
+    if(!ballotCnt || ballotCnt === localBallotCnt) return
+
+    for(localBallotCnt += 1; localBallotCnt <= ballotCnt; localBallotCnt++) {
+      await this.getBallotBasicOriginData(localBallotCnt)
+      await this.getBallotMemberOriginData(localBallotCnt)
+    }
+  }
+
+  async modifyBallotData() {
+    let voteLength = await contracts.governance.govInstance.methods.voteLength.call() // need to edit contract
+    if(!voteLength || voteLength === this.data.voteLength) return
+
+    for(this.data.voteLength += 1; this.data.voteLength <= voteLength; this.data.voteLength++) {
+      const ballotId = await contracts.ballotStorage.ballotStorageInstance.methods.getVote(this.data.voteLength).call() // need to edit contract
+      await this.getBallotBasicOriginData(ballotId)
+      await this.getBallotMemberOriginData(ballotId)
+    }
+    this.data.voteLength -= 1
+  }
+
+  async initAuthorityData() {
+    let authorityOriginData = await util.getAuthorityLists(
       constants.authorityRepo.org,
       constants.authorityRepo.repo,
       constants.authorityRepo.branch,
       constants.authorityRepo.source
     )
-    Object.keys(this.data.authorityOriginData).forEach((index) => {
-      this.data.authorityOriginData[index].addr = web3Instance.web3.utils.toChecksumAddress(this.data.authorityOriginData[index].addr)
-    })
+    this.data.authorityOriginData = await this.refineAuthority(authorityOriginData)
+    util.setAuthorityToLocal(this.data.authorityOriginData)
+  }
+
+  async refineAuthority (authorityList) {
+    let memberAuthority = {}
+    let index = 0
+    for (let i = 0; i < Object.keys(authorityList).length; i++) {
+      if(await contracts.governance.isMember(authorityList[i].addr)) {
```

Comment:
Did you run `npm run fix`? Except for this line, there are lines should be fixed by coding convention